cpp/tests/io/parquet_test.cpp

```diff
@@ -1484,4 +1484,88 @@ TEST_F(ParquetReaderTest, ReorderedColumns)
+
+  auto validity = cudf::test::make_counting_transform_iterator(0, [](auto i) { return i != 50; });
+
+  EXPECT_EQ(result.tbl->view().num_columns(), 3);
+
+  int32_t col0_data[] = {
+    -2354584, -190275,  8393572,  6446515,  -5687920, -1843550, -6897687, -6780385, 3428529,
+    5842056,  -4312278, -4450603, -7516141, 2974667,  -4288640, 1065090,  -9410428, 7891355,
+    1076244,  -1975984, 6999466,  2666959,  9262967,  7931374,  -1370640, 451074,   8799111,
+    3026424,  -6803730, 5098297,  -1414370, -9662848, 2499991,  658765,   8348874,  -6177036,
+    -9694494, -5343299, 3558393,  -8789072, 2697890,  -4454707, 8299309,  -6223703, -3112513,
+    7537487,  825776,   -495683,  328299,   -4529727, 0,        -9999999, 9999999};
+
+  EXPECT_EQ(result.tbl->view().column(0).size(), sizeof(col0_data) / sizeof(col0_data[0]));
+  cudf::test::fixed_point_column_wrapper<int32_t> col0(
+    std::begin(col0_data), std::end(col0_data), validity, numeric::scale_type{-4});
+  cudf::test::expect_columns_equal(result.tbl->view().column(0), col0);
+
+  int64_t col1_data[] = {29274040266581,  -17210335917753, -58420730139037,
+                         68073792696254,  2236456014294,   13704555677045,
+                         -70797090469548, -52248605513407, -68976081919961,
+                         -34277313883112, 97774730521689,  21184241014572,
+                         -670882460254,   -40862944054399, -24079852370612,
+                         -88670167797498, -84007574359403, -71843004533519,
+                         -55538016554201, 3491435293032,   -29085437167297,
+                         36901882672273,  -98622066122568, -13974902998457,
+                         86712597643378,  -16835133643735, -94759096142232,
+                         30708340810940,  79086853262082,  78923696440892,
+                         -76316597208589, 37247268714759,  80303592631774,
+                         57790350050889,  19387319851064,  -33186875066145,
+                         69701203023404,  -7157433049060,  -7073790423437,
+                         92769171617714,  -75127120182184, -951893180618,
+                         64927618310150,  -53875897154023, -16168039035569,
+                         -24273449166429, -30359781249192, 35639397345991,
+                         45844829680593,  71401416837149,  0,
+                         -99999999999999, 99999999999999};
+
+  EXPECT_EQ(result.tbl->view().column(1).size(), sizeof(col1_data) / sizeof(col1_data[0]));
+  cudf::test::fixed_point_column_wrapper<int64_t> col1(
+    std::begin(col1_data), std::end(col1_data), validity, numeric::scale_type{-5});
+  cudf::test::expect_columns_equal(result.tbl->view().column(1), col1);
+
+  /* There are floating point rounding errors in this test and this test is temporary, so I'm not
+    going to invest the time to fix this right now, but I'm leaving this data in here just in case
+    it is useful to know what is in the parquet file.
```

Comment:
This is the only test that covers `double`. Can you use something like `check_float_column` in CSV/JSON tests here?

Reply:
I didn't feel this was critical as this code path wasn't changed, but I would certainly like to test this as well. I'll check out those functions. Thanks for pointing that out.

Reply:
This isn't working out so well:
```
Value of: cudf::test::to_host<T>(col_lhs).first
Expected: contains 53 values, where each value and its corresponding value in { 7.03021e+18, 4.71462e+19, -9.42596e+19, 5.29485e+19, 4.78747e+19, 4.97616e+19, -5.01678e+19, 3.22062e+19, -3.69335e+19, -5.37954e+19, -8.78305e+19, 6.70937e+19, -5.1949e+17, -9.65247e+19, -3.6266e+19, -1.62049e+19, 1.37229e+19, -9.45154e+19, -3.71575e+19, 1.04632e+19, -8.79243e+19, 7.98638e+19, 5.87708e+19, 7.39387e+19, 5.89192e+19, 9.31259e+19, -2.65684e+19, -2.75241e+19, -6.34296e+19, -8.14838e+19, -9.62093e+19, 3.26623e+19, ... } Out-of-range
  Actual: { 7.03021e+18, 4.71462e+19, -9.42596e+19, 5.29485e+19, 4.78747e+19, 4.97616e+19, -5.01678e+19, 3.22062e+19, -3.69335e+19, -5.37954e+19, -8.78305e+19, 6.70937e+19, -5.1949e+17, -9.65247e+19, -3.6266e+19, -1.62049e+19, 1.37229e+19, -9.45154e+19, -3.71575e+19, 1.04632e+19, -8.79243e+19, 7.98638e+19, 5.87708e+19, 7.39387e+19, 5.89192e+19, 9.31259e+19, -2.65684e+19, -2.75241e+19, -6.34296e+19, -8.14838e+19, -9.62093e+19, 3.26623e+19, ... },
where the value pair (7.03021e+18, 7.03021e+18) at index #0 don't match
```

I think this is because it isn't just a standard double issue, but we manipulate the double when reading it. It comes in as a INT128 and we build it as a double with no decimal with `Int128ToDouble_rn` and then scale it with `(scale < 0) ? (d * kPow10[min(-scale, 39)]) : (d / kPow10[min(scale, 39)]);` I think this makes the resulting value significantly different if the scale is large enough. In my case the scale is 18. I have to crank the tolerance up to +- `1.0e+5` before it will pass.

Questions:
1) Is the scaling happening after converting to a double and losing precision a problem we should address?
2) Do we want to adjust the tolerance enough to make this pass?

Reply:
@vuule @devavret @harrism would it be ok to add an issue for this test to support the double reading and remove that burden from this PR? This PR is all but ready to go and we're short on time. The only thing that is really holding it up is adding a test for code that wasn't a part of this PR, but should certainly be tested.

Reply:
Now that the issue is filed, can you remove all the commented out data and code? I don't like to leave commented out code -- it just rots.